### PR TITLE
Add tunnel controller tests

### PR DESCRIPTION
**Pull request by tpantelis**
Preceded by some minor cleanup commits to simplify the tests.

### CODE REVIEW

File: pkg/controllers/tunnel/tunnel_test.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog"
+)
+
+const (
+	namespace = "submariner"
+)
+
+var _ = Describe("", func() {
+	klog.InitFlags(nil)
```
Comment (skitt) [inline pkg/controllers/tunnel/tunnel_test.go:27]: Shouldn’t this be an `init()` function?

File: pkg/controllers/tunnel/tunnel_test.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+})
+
+var _ = Describe("", func() {
```
Comment (skitt) [inline pkg/controllers/tunnel/tunnel_test.go:30]:
```suggestion
var _ = Describe("managing a tunnel", func() {
```

File: pkg/controllers/tunnel/tunnel_test.go (hunk 1 above)
Comment (tpantelis) [reply pkg/controllers/tunnel/tunnel_test.go:27]: This has been brought up in other PRs 😄 It fails if inside an `init` function or inside the main Test function. I forget the exact reason why.

File: pkg/controllers/tunnel/tunnel_test.go (hunk 2 above)
Comment (tpantelis) [reply pkg/controllers/tunnel/tunnel_test.go]: I made it "Managing tunnels".

**Pull request merged**